src/main/webapp/pagination.js

```diff
@@ -9,9 +9,152 @@ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 See the License for the specific language governing permissions and
 limitations under the License.*/
 
-// change content of page 
+
+let fileLength = Number.MAX_VALUE;
+
+// the function contains algo for fetching page 
+// and updating current page's content
 async function changePage(page) {
-    // TODO :- the function contains algo for fetching data 
-    // and updating current page's content
-    // WILL BE ADDED ALONG WITH PAGINATION
+    // change current page (page user is viewing at present) to the 
+    // value provided while calling the function 
+    updateCurrentPage(page);
+    const logs = document.getElementById(LOGS).getAttribute("aria-selected");
+    const searchString = document.getElementById(SEARCH_BAR).value;
+    const fileType = logs == "true" ? LOGS : ERRORS ;
+    // get the page no to be fetched next 
+    const fetchPage = getPageToBeFetched();
+    let fileName = document.getElementById(FILE_NAME).value;
+    // remove blank spaces from fileName
+    fileName =  fileName.trim();
+
+    // if we are on page 1 we need to reset the value of last page 
+    // else we already have requested page in window which we can simply
+    // display 
+    if(page == 1){
+        await resetLastPage(fileName, fileType);
+    } else {
+        display();
+    }
+
+    // if fetched page != -1 that means we need to fetch data either to create window(page == 1)
+    // or to maintain window(page > 1)
+    if(fetchPage != -1) {
```

Comment:
Can we have a separate function for updating window?
changePage() should simply call that function.